pkg/bff/api/v1/api.go

```diff
@@ -64,6 +65,11 @@ type StatusReply struct {
 // BFF v1 API Requests and Responses
 //===========================================================================
 
+// UpdateRolesParams contains a list of new roles for a collaborator.
+type UpdateRolesParams struct {
+	Roles []string `json:"roles"`
```

Comment:
I think this is good future-proofing - but right now a TRISA user should only have one role; do you think we should re-architect the roles to make them compositional rather than hierarchical? E.g. right now an organizational leader has all the permissions of an organizational collaborator and a TRISA admin has all the permissions of an organizational leader.

Reply:
Interesting, I didn't realize that they were hierarchical and that would definitely simplify things, but I'm also fine with keeping the current implementation in case we decide to go that route (I don't think we need to make changes to the current roles). One thing I haven't figured out yet is how to handle the multi-org case. For example, if a user is an organization leader in an organization are they also a leader in every other organization they are a part of?

Reply:
Oh, that's a good point … shoot. I think they might be the way things are currently set up.